### PR TITLE
Implement Kafka reporter

### PR DESCRIPTION
**Pull request by mijicd**
Implemented the Apache Kafka reporter. Reported measurements are published as JSON encoded strings into the topics-per-measurement fashion.

Other changes:
- Measurement's *toString* method now produces JSON strings.

### CODE REVIEW

File: cassandra-diagnostics-reporter-kafka/src/test/java/io/smartcat/cassandra/diagnostics/reporter/KafkaReporterShould.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import org.powermock.reflect.Whitebox;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+
+public class KafkaReporterShould {
```
Comment (mgobec) [inline cassandra-diagnostics-reporter-kafka/src/test/java/io/smartcat/cassandra/diagnostics/reporter/KafkaReporterShould.java]: We use typical java test naming (KafkaReporterTest)

File: cassandra-diagnostics-reporter-kafka/src/main/java/io/smartcat/cassandra/diagnostics/reporter/KafkaReporter.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    @Override
+    public void report(Measurement measurement) {
+        if (producer == null) {
+            logger.warn("Kafka producer is not initialized.");
+            return;
+        }
+
+        ProducerRecord<String, String> record =
```
Comment (mgobec) [inline cassandra-diagnostics-reporter-kafka/src/main/java/io/smartcat/cassandra/diagnostics/reporter/KafkaReporter.java]: No need for intermediate variable if its not used

File: cassandra-diagnostics-commons/src/main/java/io/smartcat/cassandra/diagnostics/Measurement.java [hunk 3]
```diff
@@ -91,20 +92,39 @@ private Measurement(final String name, final double value, final long time, fina
      * @return Measurement object
      */
     public static Measurement create(final String name, final double value, final long time, final TimeUnit timeUnit,
-            final Map<String, String> tags, final Map<String, String> fields) {
+                                     final Map<String, String> tags, final Map<String, String> fields) {
         return new Measurement(name, value, time, timeUnit, tags, fields);
     }
 
     @Override
     public String toString() {
```
Comment (mgobec) [inline cassandra-diagnostics-commons/src/main/java/io/smartcat/cassandra/diagnostics/Measurement.java:98]: toString() is too broad and you should not rely on it to return json. Make it a separate method.

File: cassandra-diagnostics-reporter-influx/pom.xml [hunk 4]
```diff
@@ -1,61 +1,61 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
```
Comment (mgobec) [inline cassandra-diagnostics-reporter-influx/pom.xml:1]: Whats wrong with formatter?

**Comment by mgobec** [comment]
@mijicd Fix build and check comments.

**Comment by mijicd** [comment]
@mgobec completed changes from comments

File: pom.xml [hunk 5]
```diff
@@ -62,7 +63,7 @@
             <organizationUrl>http://www.smartcat.io</organizationUrl>
         </developer>
         <developer>
-            <name>Nikola Invancevic</name>
+            <name>Nikola Ivancevic</name>
```
Comment (mgobec) [inline pom.xml:66]: 👍

**Comment by mgobec** [comment]
@mijicd Merging. Thanks

**Pull request merged**